server/meta_storage_service.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pingcap/kvproto/pkg/meta_storagepb"
+	"go.etcd.io/etcd/clientv3"
+)
+
+// MetaStorageServer wraps GrpcServer to provide meta storage service.
+type MetaStorageServer struct {
```

Comment:
What about moving the `metastoregeserver` to `msc`?

Reply:
done